src/cpp/src/text2image/stable_diffusion_3_pipeline.hpp

```diff
@@ -0,0 +1,576 @@
+    }
+
+    StableDiffusion3Pipeline(const std::string& root_dir, const std::string& device, const ov::AnyMap& properties) {
+        const std::string model_index_path = root_dir + "/model_index.json";
+        std::ifstream file(model_index_path);
+        OPENVINO_ASSERT(file.is_open(), "Failed to open ", model_index_path);
+
+        nlohmann::json data = nlohmann::json::parse(file);
+        using utils::read_json_param;
+
+        set_scheduler(Scheduler::from_config(root_dir + "/scheduler/scheduler_config.json"));
+
+        const std::string text_encoder = data["text_encoder"][1].get<std::string>();
+        if (text_encoder == "CLIPTextModelWithProjection") {
+            m_clip_text_encoder_1 =
+                std::make_shared<CLIPTextModelWithProjection>(root_dir + "/text_encoder", device, properties);
+        } else {
+            OPENVINO_THROW("Unsupported '", text_encoder, "' text encoder type");
+        }
+
+        const std::string text_encoder_2 = data["text_encoder_2"][1].get<std::string>();
+        if (text_encoder_2 == "CLIPTextModelWithProjection") {
+            m_clip_text_encoder_2 =
+                std::make_shared<CLIPTextModelWithProjection>(root_dir + "/text_encoder_2", device, properties);
+        } else {
+            OPENVINO_THROW("Unsupported '", text_encoder, "' text encoder type");
+        }
+
+        // TODO: text_encoder_3
+
+        const std::string vae = data["vae"][1].get<std::string>();
+        if (vae == "AutoencoderKL") {
+            m_vae_decoder = std::make_shared<AutoencoderKL>(root_dir + "/vae_decoder", device, properties);
+        } else {
+            OPENVINO_THROW("Unsupported '", vae, "' VAE decoder type");
+        }
+
+        const std::string transformer = data["transformer"][1].get<std::string>();
+        if (transformer == "SD3Transformer2DModel") {
+            m_transformer = std::make_shared<SD3Transformer2DModel>(root_dir + "/transformer", device, properties);
+        } else {
+            OPENVINO_THROW("Unsupported '", transformer, "'Transformer type");
+        }
+
+        // initialize generation config
+        initialize_generation_config(data["_class_name"].get<std::string>());
+    }
+
+    StableDiffusion3Pipeline(const CLIPTextModelWithProjection& clip_text_model_1,
+                             const CLIPTextModelWithProjection& clip_text_model_2,
+                             const SD3Transformer2DModel& transformer,
+                             const AutoencoderKL& vae_decoder)
+        : m_clip_text_encoder_1(std::make_shared<CLIPTextModelWithProjection>(clip_text_model_1)),
+          m_clip_text_encoder_2(std::make_shared<CLIPTextModelWithProjection>(clip_text_model_2)),
+          m_vae_decoder(std::make_shared<AutoencoderKL>(vae_decoder)),
+          m_transformer(std::make_shared<SD3Transformer2DModel>(transformer)) {}
+
+    void reshape(const int num_images_per_prompt,
```

Comment:
I've tried to create pipeline and call reshape:
```
    ov::genai::Text2ImagePipeline pipe(models_path);
    pipe.reshape(1, 1920, 1072, 7.5f);
    pipe.compile(device, ov::hint::inference_precision(ov::element::f32));

    ov::Tensor image = pipe.generate(prompt);
```
It fails with:
```
Exception from src/inference/src/cpp/infer_request.cpp:74:
Exception from src/inference/src/cpp/infer_request.cpp:66:
Exception from src/plugins/intel_cpu/src/infer_request.cpp:385:
Can't set the input tensor with index: 0, because the model input (shape=[2,77]) and the tensor (shape=(1.77)) are incompatible
```

Reply:
fixed, tested with:
```cpp
  ov::genai::Text2ImagePipeline pipe(models_path);
  pipe.reshape(1, 1920, 1072, 7.5f);
  pipe.compile(device);
  ov::Tensor image = pipe.generate(prompt, 
                                   ov::genai::height(1920),
                                   ov::genai::width(1072),
                                   ov::genai::guidance_scale(7.5f),
                                   ov::genai::num_images_per_prompt(1));
```